### PR TITLE
Improve doc

### PR DESCRIPTION
**Pull request by jonpsy**
@gf712  @vigsterkr , Any fixes recommended??

### CODE REVIEW

File: doc/readme/EXAMPLES.md [hunk 1]
```diff
@@ -35,13 +35,16 @@ Alternatively, you see [INTERFACES.md](INTERFACES.md) on how to run them manuall
 For details, see `CMakeLists.txt` in `examples/meta/` for details, `generate.py` and `translate.py` in `examples/meta/generator/`.
 
 ## Adding new examples
-It is extremely simple to add a new example: simple create another `*.sg` file.
-We are currently porting all existing Python examples in the deprecated folder `examples/undocumented/python_modular` to the new system -- a copy paste [entrance task](https://github.com/shogun-toolbox/shogun/issues/3555).
+It is extremely simple to add a new example: simple create another `*.sg.in` file in the `examples/meta/src/*/` directory. For example, if you are porting `kernel-linear.py` then you should create `linear.sg.in` under `examples/meta/src/kernel/`. To create corresponding `.sg` file it is essential to run `cmake .. && make` and then port the examples using `make meta_examples`, please ensure `BUILD_META_EXAMPLES=ON` is set. We are currently porting all existing Python examples in the deprecated folder `examples/undocumented/python_modular` to the new system -- a copy paste [entrance task](https://github.com/shogun-toolbox/shogun/issues/3555).
```
Comment (gf712) [inline doc/readme/EXAMPLES.md]:
```suggestion
It is extremely simple to add a new example: simply create another `*.sg.in` file in the `examples/meta/src/*/` directory.
For example, if you are porting `kernel-linear.py` then you should create `examples/meta/src/**kernel/linear.sg.in**`. To create the corresponding `.sg` file (which will be located in the build directory) run `cmake .. && cmake --build . --target meta_examples`. Please ensure `BUILD_META_EXAMPLES=ON` is set. 
We are currently porting all existing Python examples in the deprecated folder `examples/undocumented/python_modular` to the shogun's meta language. For further details have a look at #3555.
```

Comment (gf712) [reply doc/readme/EXAMPLES.md]: @NanuSai when writing docs you shouldn't assume that people are using `make`, some of us use `ninja` or the visual studio build system. The generic way to do it with cmake is `cmake --build . --target TARGET`.

File: doc/readme/EXAMPLES.md [hunk 2]
```diff
@@ -35,13 +35,18 @@ Alternatively, you see [INTERFACES.md](INTERFACES.md) on how to run them manuall
...
+We are currently porting all existing Python examples in the deprecated folder `examples/undocumented/python_modular` to the shogun's meta language. For further details have a look at #3555.
 
-If you are porting a new example, great! Better yet, add the [data](https://github.com/shogun-toolbox/shogun-data) so that we can do integration tests automatically as described in [DEVELOPING.md](DEVELOPING.md#testing).
+If you are porting a new example, great! Better yet, add the corresponding data. After running make meta_examples, `*.dat` file is produced by default under the `build/meta/cpp/*/` folder. This file is used for test integration and should be copied to your `data/` submodule under `testsuite/meta/*/` directory. Once that is done send a pull request to the [shogun-data repository](https://github.com/shogun-toolbox/shogun-data). Once that is done, finalize your changes a send a pull request from your main directory.
```
Comment (gf712) [inline doc/readme/EXAMPLES.md]:
```suggestion
If you are porting a new example, great! Better yet, add the corresponding data. After running make meta_examples, a `*.dat` file is produced, which by default is under the `build/meta/cpp/*/` folder. This file is used for test integration and should be copied to your `data/` submodule under `testsuite/meta/*/` directory. Once that is done send a pull request to the [shogun-data repository](https://github.com/shogun-toolbox/shogun-data). Once that is done, finalize your changes a send a pull request from your main directory. 
```

Comment (gf712) [reply doc/readme/EXAMPLES.md]: isn't the part that you added already described somewhere else?

Comment (jonpsy) [reply doc/readme/EXAMPLES.md]: I don't think so. Atleast I wasn't able to find it, but I guess it's good if it's posted right here at the EXAMPLES.md.

EDIT: Sorry it was mentioned, it was only mentioned "please follow similar instructions as original DEVELOPMENT.md" but for shogun-data repository. But I guess this simplifies things? :)

File: doc/readme/EXAMPLES.md [hunk 3]
```diff
@@ -35,13 +35,18 @@ Alternatively, you see [INTERFACES.md](INTERFACES.md) on how to run them manuall
...
+We are currently porting all existing Python examples in the deprecated folder `examples/undocumented/python_modular` to the shogun's meta language. For further details have a look at #3555.
 
-If you are porting a new example, great! Better yet, add the [data](https://github.com/shogun-toolbox/shogun-data) so that we can do integration tests automatically as described in [DEVELOPING.md](DEVELOPING.md#testing).
+If you are porting a new example, great! Better yet, add the corresponding data. After running make meta_examples, a `*.dat` file is produced, which by default is under the `build/meta/cpp/*/` folder. This file is used for test integration and should be copied to your `data/` submodule under `testsuite/meta/*/` directory. Once that is done send a pull request to the [shogun-data repository](https://github.com/shogun-toolbox/shogun-data). Once that is done, finalize your changes a send a pull request from your main directory.
```
Comment (karlnapf) [inline doc/readme/EXAMPLES.md]: "make meta_examples" should be formatted as code

**Comment by karlnapf** [comment]
I think this is useful as it makes the process more elaborate, but it needs a rebase

**Comment by jonpsy** [comment]
Hi @karlnapf thanks for the comment, I'll rebase it. 

On a side note how do i contact you? I need to discuss about the fast kernel approximate technique. Many thanks!

**Comment by karlnapf** [comment]
via the mailing list or irc (though I am not there often atm)

**Comment by jonpsy** [comment]
Rebased it :) @karlnapf Merge when you're free
Update : Wait I'll do some more fixatures as suggest by gil

File: doc/readme/EXAMPLES.md [hunk 4]
```diff
@@ -35,13 +35,21 @@ Alternatively, you see [INTERFACES.md](INTERFACES.md) on how to run them manuall
...
+ ```
+ cmake .. && cmake --build . --target meta_examples
+ ```
+Please ensure `BUILD_META_EXAMPLES=ON` is set. We are currently porting all existing Python examples in the deprecated folder `examples/undocumented/python_modular` to the shogun's meta language. For further details have a look at [#3555](https://github.com/shogun-toolbox/shogun/issues/3555).
 
-If you are porting a new example, great! Better yet, add the [data](https://github.com/shogun-toolbox/shogun-data) so that we can do integration tests automatically as described in [DEVELOPING.md](DEVELOPING.md#testing).
+If you are porting a new example, great! Better yet, add the corresponding data. After building meta_examples a `*.dat` file is produced, which by default is under the `build/meta/cpp/*/` folder. This file is used for test integration and should be copied to your `data/` submodule under `testsuite/meta/*/` directory. Once that is done send a pull request to the [shogun-data repository](https://github.com/shogun-toolbox/shogun-data). Once that is done, finalize your changes a send a pull request from your main directory.
```
Comment (karlnapf) [inline doc/readme/EXAMPLES.md]: Question, the old text referred to the developing readme. This was done to minimise redundancy, so either move the description from there to here or keep on only referring to it?

File: doc/readme/EXAMPLES.md [hunk 5]
```diff
@@ -35,13 +35,21 @@ Alternatively, you see [INTERFACES.md](INTERFACES.md) on how to run them manuall
...
+If you are porting a new example, great! Better yet, add the corresponding data. After building meta_examples a `*.dat` file is produced, which by default is under the `build/meta/cpp/*/` folder. This file is used for test integration and should be copied to your `data/` submodule under `testsuite/meta/*/` directory. Once that is done send a pull request to the [shogun-data repository](https://github.com/shogun-toolbox/shogun-data). Once that is done, finalize your changes a send a pull request from your main directory. 
 
 Please take inspiration from the existing examples, especially those that were written as part of the [Google Summer of Code](https://github.com/shogun-toolbox/shogun/wiki/GSoC-follow-up-blog-posts) 2016.
 
+A recent [pull request](https://github.com/shogun-toolbox/shogun/pull/4898), with corresponding data [pull request](https://github.com/shogun-toolbox/shogun-data) could serve as a great example highlighting the mishaps and the general procedure of example porting, use it as a reference.
+
+NOTE: As happened in the above case, please make sure to rebase both the submodule and main directory before making a PR.
```
Comment (karlnapf) [inline doc/readme/EXAMPLES.md]: Remove this is unrelated

**Review by karlnapf — COMMENTED** [review]
Had another read and made two comments

File: doc/readme/EXAMPLES.md (hunk 5 above)
Comment (jonpsy) [reply doc/readme/EXAMPLES.md]: Ok

Comment (jonpsy) [reply doc/readme/EXAMPLES.md]: I just thought it would be helpful to have a reference PR

File: doc/readme/EXAMPLES.md (hunk 4 above)
Comment (jonpsy) [reply doc/readme/EXAMPLES.md]: I say we move it because it's cumbersome to switch context and then to navigate what we're looking for. Ofcourse I'm speaking from newcomer's perspective. What's your opinion?

And if neccesary we can refer "adding data" from DEVELOPING.md to EXAMPLES.md. Meaning EXAMPLES.md will have the actual written thing

Comment (karlnapf) [reply doc/readme/EXAMPLES.md]: Having the same thing twice in different files creates maintainability nightmares, so I think referring is a better idea.

Comment (jonpsy) [reply doc/readme/EXAMPLES.md]: Noted. Will do that.

**Comment by gf712** [comment]
@karlnapf can merge this?

**Comment by jonpsy** [comment]
@gf712, not yet. I'll need to do the referencing, a sec please 😁

File: doc/readme/EXAMPLES.md [hunk 6]
```diff
@@ -35,10 +35,14 @@ Alternatively, you see [INTERFACES.md](INTERFACES.md) on how to run them manuall
...
-
-If you are porting a new example, great! Better yet, add the [data](https://github.com/shogun-toolbox/shogun-data) so that we can do integration tests automatically as described in [DEVELOPING.md](DEVELOPING.md#testing).
+It is extremely simple to add a new example: simply create another `*.sg.in` file in the `examples/meta/src/*/` directory.
+For example, if you are porting `kernel-linear.py` then you should create `examples/meta/src/**kernel/linear.sg.in**`. To create the corresponding `.sg` file (which will be located in the build directory) run
+ ```
+ cmake .. && cmake --build . --target meta_examples
+ ```
+Please ensure `BUILD_META_EXAMPLES=ON` is set. We are currently porting all existing Python examples in the deprecated folder `examples/undocumented/python_modular` to the shogun's meta language. For further details have a look at [#3555](https://github.com/shogun-toolbox/shogun/issues/3555).
+
+If you are porting a new example, great! Better yet, add the corresponding data. Check [DEVELOPMENT.md](DEVELOPING.md###Integration_testing_of_results) under "Integration testing of results" to learn how to add integration data after porting meta example.
```
Comment (gf712) [inline doc/readme/EXAMPLES.md:45]: isn't "under "Integration testing of results" redundant, as you already provide the respective href?

Comment (jonpsy) [reply doc/readme/EXAMPLES.md:45]: True! But when you click on the link we are directed a bit above "Integration test results" , so I added the line to show them where exactly to look. More like a safety measure

Comment (karlnapf) [reply doc/readme/EXAMPLES.md:45]: I just read the file as a whole. Almost all the things you added here are written above (cmake command, the flags, where files are located). This bloats the file. Further, if people are unable to locate the relevant section in the referenced develop readme, then all hope is lost.
Therefore, I am quite reluctant of merging this.
